python/pyiceberg/table/__init__.py

```diff
@@ -934,206 +976,630 @@ def case_sensitive(self, case_sensitive: bool) -> UpdateSchema:
+        """Move the field to before another field.
+
+        Args:
+            path: The path to the field.
+
+        Returns:
+            The UpdateSchema with the move operation staged.
+        """
+        full_name = ".".join(path) if isinstance(path, tuple) else path
+        field_id = self._find_for_move(full_name)
+
+        if field_id is None:
+            raise ValueError(f"Cannot move missing column: {full_name}")
+
+        before_full_name = (
+            ".".join(
+                before_path,
+            )
+            if isinstance(before_path, tuple)
+            else before_path
+        )
+        before_field_id = self._find_for_move(before_full_name)
+
+        if before_field_id is None:
+            raise ValueError(f"Cannot move {full_name} before missing column: {before_full_name}")
+
+        if field_id == before_field_id:
+            raise ValueError(f"Cannot move {full_name} before itself")
+
+        self._move(Move(field_id=field_id, full_name=full_name, other_field_id=before_field_id, op=MoveOperation.Before))
+
+        return self
+
+    def move_after(self, path: Union[str, Tuple[str, ...]], after_name: Union[str, Tuple[str, ...]]) -> UpdateSchema:
+        """Move the field to after another field.
+
+        Args:
+            path: The path to the field.
+
+        Returns:
+            The UpdateSchema with the move operation staged.
+        """
+        full_name = ".".join(path) if isinstance(path, tuple) else path
+
+        field_id = self._find_for_move(full_name)
+
+        if field_id is None:
+            raise ValueError(f"Cannot move missing column: {full_name}")
+
+        after_full_name = ".".join(after_name) if isinstance(after_name, tuple) else after_name
+        after_field_id = self._find_for_move(after_full_name)
+
+        if after_field_id is None:
+            raise ValueError(f"Cannot move {full_name} after missing column: {after_full_name}")
+
+        if field_id == after_field_id:
+            raise ValueError(f"Cannot move {full_name} after itself")
+
+        self._move(Move(field_id=field_id, full_name=full_name, other_field_id=after_field_id, op=MoveOperation.After))
+
+        return self
+
+    def commit(self) -> None:
+        """Apply the pending changes and commit."""
+        new_schema = self._apply()
+
+        if new_schema != self._schema:
+            last_column_id = max(self._table.metadata.last_column_id, new_schema.highest_field_id)
+            updates = (
+                AddSchemaUpdate(schema=new_schema, last_column_id=last_column_id),
+                SetCurrentSchemaUpdate(schema_id=-1),
+            )
+            requirements = (AssertCurrentSchemaId(current_schema_id=self._schema.schema_id),)
+
+            if self._transaction is not None:
+                self._transaction._append_updates(*updates)  # pylint: disable=W0212
+                self._transaction._append_requirements(*requirements)  # pylint: disable=W0212
```

Comment:
Not a problem, but note that if you run multiple schema updates in the same transaction, the transaction needs to ensure it sends only one `AssertCurrentSchemaId`. Right now that will fail but in the future we'll want to suppress the additional ones so that only the first schema ID is validated.

Reply:
Great observation. I checked and we only allow a single `AddSchemaUpdate` in a transaction. So I think we're okay for now. I've added a test to capture the behavior.